### PR TITLE
chore: normalized and updated copyright to 2023

### PR DESCRIPTION
**Pull request by fetzu**
- Normalized in-file copyright headers to "[...] This file Copyright © 20YY-20YY [...]" where appropriate
- Updated all the copyright end dates to 2023 (headers and COPYING) where appropriate

Note: I am working on a GitHub workflow/action (using [FantasticFiasco's action-update-license-year](https://github.com/marketplace/actions/update-license-copyright-year-s)) to automatically do this on January 1st of each year; but I have run into some issues do to some files having other copyright notices (single year, "no copyright"...) and the action just giving up once it comes across one of these. If you think that this would be valuable, I could spend some time looking into a possible solution.

### CODE REVIEW

**Comment by ckerr** [comment]
CI failure is unrealated to PR

>   Failed to perform checkout on agent: /usr/local/bin/git -c core.askpass=/home/user/buildAgent/temp/buildTmp/pass14797792754151469315 -c credential.helper= -c credential.helper=/home/user/buildAgent/temp/buildTmp/credHelper10082592072194176086.sh checkout -q -f cf3db1bfe8b33915d50405dfbdf34bac335518c0 command failed.
  exit code: 128
  stderr: error: unable to write file docs/Release-Notes.md
  error: unable to write file docs/Web-Interface.md
  error: unable to write file docs/Why-Are-My-Speeds-So-Slow.md
  error: unable to write file docs/Why-is-my-port-closed.md
  fatal: cannot create directory at 'docs/resources': No space left on device

CC @mikedld

**Comment by ckerr** [comment]
>  If you think that this would be valuable, I could spend some time looking into a possible solution.

It's not urgent but if you want to put time into it I'd be happy to reap the rewards of your effort :smile_cat:

**Comment by ckerr** [comment]
Sanity CI failure is unrelated to the PR:

> [12:46:21 ](https://trtc.mikedld.com/buildConfiguration/Transmission_Sanity_DragonFlyBSDX8664/340195?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandchucktcpluginBuild=true&expandPull+Request+Details=true&expandBuildProblemsSection=true&expandBuildChangesSection=true&showLog=340195_76_76&logView=flowAware)  Failed to perform checkout on agent: /usr/local/bin/git -c core.askpass=/home/user/buildAgent/temp/buildTmp/pass14797792754151469315 -c credential.helper= -c credential.helper=/home/user/buildAgent/temp/buildTmp/credHelper10082592072194176086.sh checkout -q -f cf3db1bfe8b33915d50405dfbdf34bac335518c0 command failed.
  exit code: 128
  stderr: error: unable to write file docs/Release-Notes.md
  error: unable to write file docs/Web-Interface.md
  error: unable to write file docs/Why-Are-My-Speeds-So-Slow.md
  error: unable to write file docs/Why-is-my-port-closed.md
  fatal: cannot create directory at 'docs/resources': No space left on device

**Review by ckerr — APPROVED** [review]

File: gtk/FilterListModel.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// This file Copyright © 2023 Mnemosyne LLC.
+// This file Copyright © 2023-2023 Mnemosyne LLC.
```
Comment (Coeur) [inline gtk/FilterListModel.h:1]: I don't think this was appropriate.

**Review by Coeur — COMMENTED** [review]
My feeling is that instead of updating the year each year, we should just delete the years values which are unneeded in a context of Git versioning.

Aka, first line should just be one of:
>// This file Copyright © Mnemosyne LLC.

>// This file Copyright © Juliusz Chroboczek.

>// This file Copyright © Transmission authors and contributors.

File: gtk/FilterListModel.h (hunk 1 above)
Comment (fetzu) [reply gtk/FilterListModel.h:1]: My mistake, the script ended up adding an ending date. This will be fixed with a upcoming PR.

**Comment by fetzu** [comment]
@Coeur: I agree, this might be an easier solution in the long term and does make sense in the context of a git repo.

I'll be submitting two PRs shortly:
1. The first one to standardize all the copyright headers across the repo (except for /third-party), removing the dates and leaving "// This file Copyright © {person}".
2. A second one to take care of some isolated instances where dates are shown (COPYING, GUI) via a github workflow/action.

**Comment by Coeur** [comment]
@fetzu note that in your logic/process, you forgot a few files:
https://github.com/transmission/transmission/blob/b7099270b6bbcd9181b5b42da1e5a39ccdf182ae/libtransmission/cache.cc#L1

https://github.com/transmission/transmission/blob/b7099270b6bbcd9181b5b42da1e5a39ccdf182ae/libtransmission/completion.cc#L1

https://github.com/transmission/transmission/blob/b7099270b6bbcd9181b5b42da1e5a39ccdf182ae/libtransmission/file.h#L1

https://github.com/transmission/transmission/blob/b7099270b6bbcd9181b5b42da1e5a39ccdf182ae/libtransmission/history.h#L1

etc. etc.

**Pull request merged**